Logout

fuihfuefuiewn opened the pull request:
"Ushbu pull request faqat logoutni o'z ichiga oladi, aslida HTML juda befoyda, chunki deyarli hech narsa bo'lmasligi kerak, faqat chiqib ketish va baxtli bo'lish... Mariano matritsalari kabi."

Sa4dUs committed:
Authentication and User related frontend features (#26)

* Add global ignore for eslint

This solves long lint times

* Lint hotfix

* En equipo todo va bien

Co-Authored-By: fuihfuefuiewn <131182551+fuihfuefuiewn@users.noreply.github.com>
Co-Authored-By: Marcelo Domínguez <69964857+Sa4dUs@users.noreply.github.com>
Co-Authored-By: Jorge Cimadevilla <109004232+Lamoara@users.noreply.github.com>

* Update gitignore

Co-Authored-By: fuihfuefuiewn <131182551+fuihfuefuiewn@users.noreply.github.com>
Co-Authored-By: Marcelo Domínguez <69964857+Sa4dUs@users.noreply.github.com>
Co-Authored-By: Jorge Cimadevilla <109004232+Lamoara@users.noreply.github.com>

* Path fixed

---------

Co-authored-by: Alejandro García Prada <114813960+AlexGarciaPrada@users.noreply.github.com>
Co-authored-by: fuihfuefuiewn <131182551+fuihfuefuiewn@users.noreply.github.com>
Co-authored-by: Jorge Cimadevilla <109004232+Lamoara@users.noreply.github.com>

Sa4dUs committed:
Backend rollup into `main` (#27)

* created remaining services

* added services to compose

* ipdated workflows to use all services

* Just for drafting

* API infra setup

* Add parser middleware

Note(Sa4dUs): middleware logic can prob be improved for more convenient middleware creation.

* Setup basic forwarding

Note(Sa4dUs): I regressed `types.rs`, as this is a generic api gateway, think it's better to KISS

Note(Sa4dUs): `handler: > handler` logic is pretty messy and not well organised, will update in following commits, but i wanted to get it working first.

* Init middleware-based routing pipeline

* Database Creation and Connection

Problems with formattingDatabase Creation and Connection

Problems with formatting

* Estructure Redifining

* Some db operations

* Add routing middleware

TODO(Sa4dUs): check if we can improve this implementation from the example [here](https://docs.rs/axum/latest/axum/middleware/index.html#rewriting-request-uri-in-middleware)

* Check `Authorization` header to contain a valid JWT token

* Setup fluvio

* Setup fluvio topic structs

* Setup base structure

* General setup

* temp fix

* This is not compiled, probably it doesn't

* Errors fixing

* refactor and friend requests

* Now i know how to use requests

* Register flow

* Make config environment specific

* Create crate specific .gitignore

* Setup simple testing suite

* refactor based on comments

* Minor adjustments

* Add CORS layer

* Update .env.example for CORS

* .env fix in compose

* sign_in flow

* Little fixes until

* Changed arc to State and other db fixes

* Fluvio

* Some fixes

* More fixes

* This?

* Minor fixes

NOTE(Sa4dUs): Adding `.clone` till all the issues are solver may not be the best way to handle this, but it's fine for now

* Friend request and refactor

* format fix

* get blocks

* fmt fix

* Fixes following Sa4dus comments

* Last fixes??

* Fixing the router order

* Updating dependencies and .env

* Fixing dependencies

* remaining routes added

* cleanup

* fix gitignore

* sql fixes

* small fixes

* Auth-service update

* removed allow dead code

* removed testing users

* comment

* remove testing

* remove test

* fix

* Separating db errors

* Minor fix

* Some tracing that I missed

* Deleted Jorge's conflicted file

* Minor fixes

Improve config setup for api

* Update backend-deploy workflow

* User service naming hotfix

* sql migrations user-service hotfix

* sql migrations auth-service hotfix

* fluvio auth-service hotfix

* Cleanup

* Initial setup

* ws handler

* Added fluvio topics

* CORS hotfix

* Refactor evend and friendship events

* Fluvio topics fixed

* CORS hotfix

* Added fluvio topics

* Refactor evend and friendship events

* Fluvio topics fixed

* Fluvio readers fn setup

* Prevent preflight from being rejected due to layer order

* Fluvio reader rework

* Basic implementation

* _

* Minor updates

* Move load balancer logic to a dedicated middleware

* Improve logging

* Allow any methods

* Minor updates

* Full websocket proxy

WIll revisit implementation probably

* Add query params support

* Apply Strategy pattern for load balancing

* Impl rate limiter layer

* Minor fixes from previous PR

* Setup circuit breaking

* Integrate circuit breaker in handler pipeline

* Fixed compose to work locally

* _

* Fix `record_key` not being parsed to string properly

* _

* Add subprotocol-based JWT authentication

You can now authenticate using the builtin header:
```
Sec-Websocket-Protocol: JWT
```

Javascript snippet:
```
let socket = new WebSocket(URL, [JWT_TOKEN])
```

---------

Co-authored-by: Lamoara <jorginos40@gmail.com>
Co-authored-by: Alejandro García Prada <114813960+AlexGarciaPrada@users.noreply.github.com>
Co-authored-by: Jorge Cimadevilla <109004232+Lamoara@users.noreply.github.com>

fuihfuefuiewn committed:
logOut

ESPAÑAAAA

Lint

Importante

fuihfuefuiewn committed:
logOut

fuihfuefuiewn committed:
Importante

fuihfuefuiewn committed:
Lint

fuihfuefuiewn committed:
ESPAÑAAAA

Sa4dUs commented on frontend/src/app/logout/logOut.component.html:
```diff
@@ -0,0 +1,7 @@
+<h2>LogOut</h2>
+
+<button (click)="onSubmitLogOut()" [disabled]="loading">
+  {{ loading ? "Cerrando sesión..." : "Cerrar sesión" }}
```
Translate this to english at least XD

AlexGarciaPrada replied:
In english please

Sa4dUs commented on frontend/src/app/logout/logOut.component.ts:
```diff
@@ -0,0 +1,49 @@
+        const user = JSON.parse(localStorage.getItem("user") || "{}");
+
+        if (!user?.username) {
+            console.warn("Messages in spanish as Gibraltar.");
```
Make this comment serious and useful pls

fuihfuefuiewn replied:
What could be more serious than trying to denounce the existence of a colonial remnant recognized by the UN and currently undergoing decolonization, to make matters worse.

AlexGarciaPrada commented on frontend/src/app/logout/logOut.component.ts:
```diff
@@ -0,0 +1,49 @@
+        const user = JSON.parse(localStorage.getItem("user") || "{}");
+
+        if (!user?.username) {
+            console.warn("Messages in spanish as Gibraltar.");
```
???

fuihfuefuiewn replied:
lo primero, la parte final sin eso peta.
lo segundo, funciona igual sin el ?, solo es porque si no existe es un null.null y podría no existir (evitar que te aparezca un mensaje rojo en Narnia). De todas formas, lo tienes sin eso y con eso en diferentes partes de mi coso

Sa4dUs commented on frontend/src/app/logout/logOut.component.ts:
```diff
@@ -0,0 +1,49 @@
+                next: () => {
+                    localStorage.removeItem("token");
+                    localStorage.removeItem("user");
+                    console.log("You are fired");
```
Same as above

Sa4dUs commented on frontend/src/app/logout/logOut.component.ts:
```diff
@@ -0,0 +1,49 @@
+                    localStorage.removeItem("token");
+                    localStorage.removeItem("user");
+                    console.log("You are fired");
+                    this.router.navigate(["/login"]); //suena razonable que cuando hagas logOut te redirija aquí
```
Code in english

AlexGarciaPrada commented on frontend/src/app/logout/logOut.component.ts:
```diff
@@ -0,0 +1,49 @@
+                    localStorage.removeItem("token");
+                    localStorage.removeItem("user");
+                    console.log("You are fired");
+                    this.router.navigate(["/login"]); //suena razonable que cuando hagas logOut te redirija aquí
```
Agree,  there or home

Sa4dUs replied:
You cannot visit home unless you're authenticated, right? I think `/login` is fine

fuihfuefuiewn replied:
vale, creo que sino escribo nada  no deja escurrir el bulto.
PEZ. Ya lo hablamos hace una semana esto....por ahora si puedes, pero te da un mensaje de error (por si acaso está así)

Sa4dUs commented on frontend/src/app/personal-user/user/user.component.html:
```diff
@@ -9,4 +9,5 @@ <h2>Mi querida cuenta</h2>
 <a routerLink="/friendship-friend">Ver Lista de Amigos</a>
```
Translate this to english too pls, @AlexGarciaPrada forgot to do that in his PR

Sa4dUs commented on frontend/src/environment/environment.secret.ts:
```diff
@@ -18,4 +18,4 @@
 
 /* Fill this with actual values */
-export const SERVER_ROUTE = "";
+export const SERVER_ROUTE = "https://lamoara.duckdns.org/devcord";
```
Don't expose server url on the code, this file should be ignored by git. Rebase to also remove it from the commit history

AlexGarciaPrada commented on frontend/src/environment/environment.secret.ts:
```diff
@@ -18,4 +18,4 @@
 
 /* Fill this with actual values */
-export const SERVER_ROUTE = "";
+export const SERVER_ROUTE = "https://lamoara.duckdns.org/devcord";
```
This shouldn't be here

fuihfuefuiewn replied:
Exclamó la dulce princesa

AlexGarciaPrada commented on frontend/src/app/logout/logOut.component.ts:
```diff
@@ -0,0 +1,49 @@
+                },
+                error: (error) => {
+                    console.error(
+                        this.errorsMap.getErrorMessage(context, error),
```
Have you changed the json with the codes

fuihfuefuiewn replied:
justifica tu respuesta

fuihfuefuiewn committed:
Revert "Merge branch 'main' into front/cosasDeLaLista"

This reverts commit 6be43bb9f8ce760da2bafec732155fe8263f259e, reversing
changes made to 7e3da39845e7d7112c99b480e3231058a90b479d.

fuihfuefuiewn committed:
Reapply "Merge branch 'main' into front/cosasDeLaLista"

This reverts commit 83d8c60939bb48803efdc5e5ca9d421a9aa7526e.

fuihfuefuiewn committed:
Revert "Reapply "Merge branch 'main' into front/cosasDeLaLista""

This reverts commit 03f2118c9f07538bb4f147c6d49f8893d6e7060a.

fuihfuefuiewn committed:
Cambio para el warning

No sé si es eso por lo que """falla"""

fuihfuefuiewn committed:
Prueba y error

Parte 1

fuihfuefuiewn committed:
Revert "Prueba y error"

This reverts commit 68e915d4def79283ced411d17e0843fb4d0e40c6.

fuihfuefuiewn committed:
Prueba y error 2

fuihfuefuiewn committed:
Buen commit mejor comentario

COSAS:
-Eliminar el clon malvado de user
-Apelar a mantener la sesión iniciada
-Logout reformado

fuihfuefuiewn committed:
Cambios del lint

fuihfuefuiewn committed:
Primera Fase de la solución

Gran problema, pero primer paso dado. Ya la parte de la duplicidad está fuera
